src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2767,3 +2778,209 @@ class CreateSchemaStatementSegment(BaseSegment):
             optional=True,
         ),
     )
+
+
+@tsql_dialect.segment()
+class MergeStatementSegment(BaseSegment):
+    """`MERGE` statement.
+
+    https://docs.microsoft.com/en-us/sql/t-sql/statements/merge-transact-sql?view=sql-server-ver15
+    """
+
+    type = "merge_statement"
+
+    match_grammar = Sequence(
+        "MERGE",
+        Sequence(
+            "TOP",
+            OptionallyBracketed(Ref("ExpressionSegment")),
+            Sequence("PERCENT", optional=True),
+            optional=True,
+        ),
+        Sequence("INTO", optional=True),
+        Indent,
+        OneOf(
+            Ref("TableReferenceSegment"),
+            Ref("AliasedTableReferenceGrammar"),
+            Sequence(
+                Ref("TableReferenceSegment"),
+                Ref("PostTableExpressionGrammar", optional=True),
+                Ref("AliasExpressionSegment", optional=True),
+            ),
+        ),
+        Dedent,
+        "USING",
+        Indent,
+        OneOf(
+            Sequence(
+                Ref("TableReferenceSegment"),
+                Ref("AliasExpressionSegment", optional=True),
+            ),
+            Sequence(
+                OptionallyBracketed(
+                    Ref("UnorderedSelectStatementSegment"),
+                ),
+                Ref("AliasExpressionSegment", optional=True),
+                Ref("BracketedColumnReferenceListGrammar", optional=True),
+            ),
+        ),
+        Dedent,
+        Ref("JoinOnConditionSegment"),
+        AnyNumberOf(
+            Ref("MergeMatchedClauseSegment"),
+            Ref("MergeNotMatchedClauseSegment"),
+        ),
```

Comment:
Can we throw a min_times=1 in here?  I believe there needs to be at least one of these clauses.

https://docs.microsoft.com/en-us/sql/t-sql/statements/merge-transact-sql?view=sql-server-ver15#remarks

"At least one of the three MATCHED clauses must be specified, but they can be specified in any order."

Reply:
Added as suggested, thanks